test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpectContinueTest.java

```diff
@@ -419,8 +422,12 @@ public void testFixedLengthRequestWithDoubleExpect100ContinueResponse() throws E
     }
 
     // Creates a connection with all the common settings used in each test
-    private HttpURLConnection createConnection() throws IOException {
-        URL url = new URL("http://localhost:" + control.serverSocket.getLocalPort());
+    private HttpURLConnection createConnection() throws Exception {
+        URL url = URIBuilder.newBuilder()
+                .scheme("http")
+                .host(InetAddress.getLoopbackAddress())
```

Comment:
This will change the call flow of the test in a slightly subtle way. In the original the "host" is supplied i.e. localhost which should map to the loopback address, and in the change the loopback IP address is being supplied diectly. In terms of equivalence then supplying a host string might be more appropriate:
.host("localhost")

Reply:
No that's precisely what we want to avoid. Because how "localhost" maps to an InetAddress depends on the machine configuration, which is a recipe for intermittent failures.

I'd suggest:
```
        URL url = URIBuilder.newBuilder()
                .scheme("http")
                .localhost()
```
(which is actually the same)

Reply:
At a glance there isn't a `.localhost()` only a `.loopback()` is that what you were referring to?